dissect/target/plugins/filesystem/ntfs/mft.py

```diff
@@ -138,53 +145,62 @@ def mft(self, compact: bool = False):
         else:
             record_formatter = formatter
 
-        for fs in self.target.filesystems:
-            if fs.__type__ != "ntfs":
-                continue
-
-            # If this filesystem is a "fake" NTFS filesystem, used to enhance a
-            # VirtualFilesystem, The driveletter (more accurate mount point)
-            # returned will be that of the VirtualFilesystem. This makes sure
-            # the paths returned in the records are actually reachable.
-            drive_letter = get_drive_letter(self.target, fs)
-            volume_uuid = get_volume_identifier(fs)
-
+        if fs:
```

Comment:
Because the default is `-1`, if will by default evaluate to `True`. When the user passes `0`, it will evaluated to `False`.

I would also argue that the user should be able to use negative numbers, for example, when the users wants the last NTFS filesystem.

So long story short, I would suggest to set the default to None and check for `if fs is None` instead. Please then also change the type hint to `Optional[int]` for `fs`

Reply:
Fixed